File: app/collector/browser.py
from playwright.sync_api import sync_playwright

from config.settings import (
    CHROME_PATH,
    USER_DATA_DIR,
    PROFILE,
)

class Browser:

    def launch(self):

        playwright = sync_playwright().start()

        browser = playwright.chromium.launch_persistent_context(

            executable_path=CHROME_PATH,

            user_data_dir=USER_DATA_DIR,

            channel=None,

            headless=False,

            args=[
                f"--profile-directory={PROFILE}",
                "--start-maximized",
            ],

            no_viewport=True,
        )

        page = browser.new_page()

        return playwright, browser, page

File: collector/browser.js
const puppeteer = require("puppeteer-extra");

const CHROME_EXECUTABLE =
    "C:/Program Files/Google/Chrome/Application/chrome.exe";

const CHROME_USER_DATA_DIR =
    "E:/Maplestory/MSU-Market-Analyzer/chrome-profile";

class Browser {

    constructor() {
        this.browser = null;
        this.page = null;
    }

    async launch() {

        this.browser = await puppeteer.launch({
    headless: false,
    executablePath: CHROME_EXECUTABLE,
    userDataDir: CHROME_USER_DATA_DIR,

    ignoreDefaultArgs: ['--disable-extensions'],

    args: [
        '--profile-directory=Profile 3',
        '--no-sandbox',
        '--disable-setuid-sandbox',
        '--start-maximized'
    ],

    defaultViewport: null
	});

        this.page = await this.browser.newPage();

        return this.page;

    }

    async close() {

        if (this.browser) {

            await this.browser.close();

        }

    }

}

module.exports = Browser;